src/executor/select.rs

```diff
@@ -997,31 +998,96 @@ where
     L: EntityTrait,
     R: EntityTrait,
 {
-    let mut acc: Vec<(L::Model, Vec<R::Model>)> = Vec::new();
-    for (l, r) in rows {
-        if let Some((last_l, last_r)) = acc.last_mut() {
-            let mut same_l = true;
-            for pk_col in <L::PrimaryKey as Iterable>::iter() {
-                let col = pk_col.into_column();
-                let val = l.get(col);
-                let last_val = last_l.get(col);
-                if !val.eq(&last_val) {
-                    same_l = false;
+    {
+        let pkcol = <L::PrimaryKey as Iterable>::iter()
+            .next()
+            .expect("should have primary key")
+            .into_column();
+
+        let keys: Vec<Value> = rows.iter().map(|row| row.0.get(pkcol)).collect();
+        let mut unique_keys: Vec<Value> = Vec::new();
```

Comment:
We can have used a HashSet to find unique values. But I doubt this is even needed for the algorithm.